### ISSUE TITLE
Feature request. Add ay emulator to droidsound?

### ISSUE THREAD

**Issue by juanreina921** [droidmjt/Droidsound#605]
Hello droidjmt! I have a feature request. Could it be a possibility for you to add the ay emulator to droidsound? For those who don't know, ay emul/ay emulator, is a soundchip player, it can accurately play almost all forms of specxy music, including. 
ST1 - Sound Tracker v1.xx uncompiled; STC, ZXS - Sound Tracker v1.xx, Super Sonic v1.xx; PSC - Pro Sound Creator v1.xx; AS0, ASC - ASC Sound Master v0.xx-2.xx; PT1, PT2, PT3 - Pro Tracker v1.xx-3.xx, Vortex Tracker II v1.0; STP - Sound Tracker Pro; FTC - Fast Tracker v1.xx; FLS - Flash Tracker; SQT - SQ-Tracker; GTR - Global Tracker v1.x; FXM - Fuxoft AY Language; AY of AMAD subtype - Amadeus modules, FXM analog; PSM - Pro Sound Maker. It can also play ZX Spectrum's, Atari ST/STe's or Amstrad CPC's memory dumps with player for processor (Z80 or MC68000) like: AY of EMUL subtype (DeliAY and AYPlay projects files); AYM (RDOSPLAY project files); SNDH (Atari ST or Atari STe music files). In adission, its ym playback far outshines the likes of the plughin we have rite now, as it plays this song perfectly! []()ftp://modland.com/pub/modules/YM/Tao/dancer.ym other songs from Tao and others use this querk in the soundchip that the plughin we have rite now , doesn't emulate. I hope this gets added soon, as I would love to listen to these and much more files playing correctly in droidsound. 
Source code of laytest and stabil []()https://bulba.untergrund.net/Ay_Emul29b26.src.7z
Thanks again in advance, and keep up the good work with this grate app!

**Comment by droidmjt** [comment]
Those formats you mentioned are already supported.

**Comment by juanreina921** [comment]
Then could you make ym playback be done via the player we have now? Ym output is still being played by Plugin: STSound, whitch was good back in it's day, but it's becoming obcelete and it doesn't have the ym capabilities like newer player plugins like ay emulator and others. Might I suggest listening to dancer via ay emulator and then via droidsound again?

**Comment by droidmjt** [comment]
I can add a switch to play YM files either with STSound or ZXTune plugins, but I suggest you forget the YM format completely and listen to SNDH files instead.

**Comment by juanreina921** [comment]
A switch would be nice to have, the hard thing is, ym is not werth dropping as some games, like robocop, were only in ym format.

**Comment by droidmjt** [comment]
https://www.dropbox.com/s/4xzhd1hy7ajy1bz/droidsounde167_241_32bit.apk?dl=1
https://www.dropbox.com/s/7wg3w90ld5uvkgv/droidsounde167_241_64bit.apk?dl=1
test it

**Comment by juanreina921** [comment]
@Droidjmt I'm so sorry to be bugging you with this issue, even after you went through the trouble of adding a switch, It looks like zxtune plugin fails more at emulating some of the ay and ym capabilities. And when I try loading this song made by Tao, an sndh, I hear a pop from the chip, and nothing plays. Here is the file []()http://www.fileconvoy.com/dfl.php?id=g7d001628987596d11000297394ad334a9a6adf1a65 
I tried pressing stop and then play again, and droidsound reports too many failyors, stopping. Reextracting the file again and adding it into droidsound has the same result.  I also did a clean reinstall, clearing the cash, data uninstalling, deleting the folder and then installing again, but it still doesn't play. Thanks in advance for any updates, and sorry again for bugging you on this issue.
*Edit. Don't worry about the file, it will be there for over two weeks.
